### PR TITLE
openthread: fix compile error when CLI is not present

### LINKED ISSUE #11498: Cannot compile OpenThread example without the CLI-APP.
#### Description
Cannot compile OpenThread without the cli example feature.

#### Steps to reproduce the issue
On an OpenThread example comment these two lines in the makefile: 
```
USEMODULE += openthread-cli-ftd
USEMODULE += openthread-cli-mtd
```
then do:
`make all`

#### Expected results
Run a Thread instance without the CLI example.

#### Actual results
If doing the procedure on a **clean** example:

```
arm-none-eabi-gcc: error: /home/lab/Documents/TESTRI/RIOT/examples/openthread/bin/samr21-xpro/openthread-ftd.a: No such file or directory
make: *** [/home/lab/Documents/TESTRI/RIOT/examples/openthread/bin/samr21-xpro/openthread.elf] Error 1
```

On the other hand, if commenting the lines in the makefile after a succesfull build with CLI example:

Compiles Ok, but the CLI example remains in the program.

#### Versions

Operating System Environment
-----------------------------
       Operating System: "Ubuntu" "14.04.6 LTS, Trusty Tahr"
                 Kernel: Linux 4.4.0-146-generic i686 i686

Installed compiler toolchains
-----------------------------
             native gcc: gcc (Ubuntu 4.9.4-2ubuntu1~14.04.1) 4.9.4
      arm-none-eabi-gcc: arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2018-q3-update) 7.3.1 20180622 (release) [ARM/embedded-7-branch revision 261907]
                avr-gcc: missing
       mips-mti-elf-gcc: missing
             msp430-gcc: missing
   riscv-none-embed-gcc: missing
                  clang: missing

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: "3.0.0"
    mips-mti-elf-newlib: missing
riscv-none-embed-newlib: missing
               avr-libc: missing (missing)

Installed development tools
---------------------------
                  cmake: cmake version 2.8.12.2
               cppcheck: missing
                doxygen: 1.8.6
                 flake8: missing
                    git: git version 1.9.1
                   make: GNU Make 3.81
                openocd: Open On-Chip Debugger 0.9.0-dev-gbdd8312 (2018-04-05-13:14)
                 python: Python 2.7.6
                python2: Python 2.7.6
                python3: Python 3.4.3
             coccinelle: missing

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR fixes compilation error when OpenThread CLI is not included (see #11498).

I separated the CLI and FTD/MTD variables so the whole logic is much simpler.

### Testing procedure
Compile and run OpenThread in FTD and MTD:
```
OPENTHREAD_TYPE=ftd make -C examples/openthread all flash term
```
and 
```
OPENTHREAD_TYPE=mtd make -C examples/openthread all flash term
```
Run `help` command in both instances. The MTD node should show less commands than the FTD

Then, comment the `USEMODULE=openthread-cli` line in `examples/openthread/Makefile`. Compile and run both FTD and MTD again. Now, the shell shouldn't be available (and after reset, you should see `Current panid: ...`)

### Issues/PRs references
Fixes #11498

### CODE REVIEW

File: pkg/openthread/Makefile [hunk 1]
```diff
@@ -3,15 +3,18 @@ PKG_URL=https://github.com/openthread/openthread.git
...
+  TD = ftd
   $(info Compile OpenThread for FTD device)
-  OPENTHREAD_ARGS += --enable-cli-app=ftd
-endif
-ifneq (,$(filter openthread-cli-mtd,$(USEMODULE)))
+else ifneq (,$(filter openthread-mtd,$(USEMODULE)))
   $(info Compile OpenThread for MTD device)
-  OPENTHREAD_ARGS += --enable-cli-app=mtd --enable-joiner
+  TD = mtd
+  JOINER = "--enable-joiner"
```
Comment (leandrolanzieri) [inline pkg/openthread/Makefile]: Are quotes needed here?

Comment (jia200x) [reply pkg/openthread/Makefile]: actually not. I will remove them

**Comment by cladmi** [comment]
Can you provide with the expected output in master and this PR ?

File: pkg/openthread/Makefile [hunk 2]
```diff
@@ -3,15 +3,18 @@ PKG_URL=https://github.com/openthread/openthread.git
...
   $(info Compile OpenThread for MTD device)
-  OPENTHREAD_ARGS += --enable-cli-app=mtd --enable-joiner
+  TD = mtd
+  JOINER = --enable-joiner
+else
+  $(error "Please use either USEMODULE=openthread-ftd or USEMODULE=openthread-mtd")
 endif
-OPENTHREAD_ARGS += --enable-application-coap
+
+OPENTHREAD_ARGS += --enable-cli-app=$(TD) $(JOINER) --enable-application-coap
```
Comment (leandrolanzieri) [inline pkg/openthread/Makefile]: Could the `--enable-cli-app` flag be removed if the CLI module is not used?

**Comment by jia200x** [comment]
@cladmi 

This is the output of `help` in a FTD device:
```
2019-05-09 18:11:19,466 - INFO # help
2019-05-09 18:11:19,467 - INFO # autostart
2019-05-09 18:11:19,468 - INFO # bufferinfo
2019-05-09 18:11:19,468 - INFO # channel
2019-05-09 18:11:19,469 - INFO # child
2019-05-09 18:11:19,470 - INFO # childmax
2019-05-09 18:11:19,471 - INFO # childtimeout
2019-05-09 18:11:19,472 - INFO # coap
2019-05-09 18:11:19,474 - INFO # contextreusedelay
2019-05-09 18:11:19,475 - INFO # counter
2019-05-09 18:11:19,476 - INFO # dataset
2019-05-09 18:11:19,478 - INFO # delaytimermin
2019-05-09 18:11:19,479 - INFO # discover
2019-05-09 18:11:19,480 - INFO # eidcache
2019-05-09 18:11:19,481 - INFO # eui64
2019-05-09 18:11:19,481 - INFO # extaddr
2019-05-09 18:11:19,482 - INFO # extpanid
2019-05-09 18:11:19,483 - INFO # factoryreset
2019-05-09 18:11:19,484 - INFO # hashmacaddr
2019-05-09 18:11:19,485 - INFO # ifconfig
2019-05-09 18:11:19,486 - INFO # ipaddr
2019-05-09 18:11:19,487 - INFO # ipmaddr
2019-05-09 18:11:19,488 - INFO # joinerport
2019-05-09 18:11:19,488 - INFO # keysequence
2019-05-09 18:11:19,489 - INFO # leaderdata
2019-05-09 18:11:19,491 - INFO # leaderpartitionid
2019-05-09 18:11:19,492 - INFO # leaderweight
2019-05-09 18:11:19,493 - INFO # masterkey
2019-05-09 18:11:19,493 - INFO # mode
2019-05-09 18:11:19,495 - INFO # networkdiagnostic
2019-05-09 18:11:19,497 - INFO # networkidtimeout
2019-05-09 18:11:19,498 - INFO # networkname
2019-05-09 18:11:19,499 - INFO # panid
2019-05-09 18:11:19,500 - INFO # parent
2019-05-09 18:11:19,501 - INFO # parentpriority
2019-05-09 18:11:19,502 - INFO # ping
2019-05-09 18:11:19,502 - INFO # pollperiod
2019-05-09 18:11:19,504 - INFO # promiscuous
2019-05-09 18:11:19,504 - INFO # pskc
2019-05-09 18:11:19,506 - INFO # releaserouterid
2019-05-09 18:11:19,507 - INFO # reset
2019-05-09 18:11:19,508 - INFO # rloc16
2019-05-09 18:11:19,509 - INFO # router
2019-05-09 18:11:19,510 - INFO # routerdowngradethreshold
2019-05-09 18:11:19,511 - INFO # routerrole
2019-05-09 18:11:19,513 - INFO # routerselectionjitter
2019-05-09 18:11:19,515 - INFO # routerupgradethreshold
2019-05-09 18:11:19,516 - INFO # scan
2019-05-09 18:11:19,517 - INFO # singleton
2019-05-09 18:11:19,518 - INFO # state
2019-05-09 18:11:19,518 - INFO # thread
2019-05-09 18:11:19,519 - INFO # txpowermax
2019-05-09 18:11:19,520 - INFO # version
```

And this is the one for a MTD device:
 ```2019-05-09 18:13:22,338 - INFO # help
2019-05-09 18:13:22,339 - INFO # autostart
2019-05-09 18:13:22,340 - INFO # bufferinfo
2019-05-09 18:13:22,341 - INFO # channel
2019-05-09 18:13:22,342 - INFO # childtimeout
2019-05-09 18:13:22,343 - INFO # coap
2019-05-09 18:13:22,344 - INFO # counter
2019-05-09 18:13:22,346 - INFO # dataset
2019-05-09 18:13:22,346 - INFO # discover
2019-05-09 18:13:22,347 - INFO # eui64
2019-05-09 18:13:22,348 - INFO # extaddr
2019-05-09 18:13:22,348 - INFO # extpanid
2019-05-09 18:13:22,349 - INFO # factoryreset
2019-05-09 18:13:22,350 - INFO # hashmacaddr
2019-05-09 18:13:22,351 - INFO # ifconfig
2019-05-09 18:13:22,352 - INFO # ipaddr
2019-05-09 18:13:22,353 - INFO # ipmaddr
2019-05-09 18:13:22,353 - INFO # joiner
2019-05-09 18:13:22,354 - INFO # keysequence
2019-05-09 18:13:22,356 - INFO # leaderdata
2019-05-09 18:13:22,357 - INFO # masterkey
2019-05-09 18:13:22,357 - INFO # mode
2019-05-09 18:13:22,358 - INFO # networkname
2019-05-09 18:13:22,359 - INFO # panid
2019-05-09 18:13:22,360 - INFO # parent
2019-05-09 18:13:22,360 - INFO # ping
2019-05-09 18:13:22,361 - INFO # pollperiod
2019-05-09 18:13:22,363 - INFO # promiscuous
2019-05-09 18:13:22,363 - INFO # reset
2019-05-09 18:13:22,364 - INFO # rloc16
2019-05-09 18:13:22,365 - INFO # scan
2019-05-09 18:13:22,365 - INFO # singleton
2019-05-09 18:13:22,366 - INFO # state
2019-05-09 18:13:22,367 - INFO # thread
2019-05-09 18:13:22,368 - INFO # txpowermax
2019-05-09 18:13:22,369 - INFO # version
```

If `openthread-cli` is not included, then you shouldn't see these commands

**Comment by taherrera** [comment]
I have tested the following **without CLI** on a ftd:

- ping to a remote desktop via BR
- udp to a remote desktop via BR
- (and all ot-api features that these points require)

thanks for the fix @jia200x

**Comment by leandrolanzieri** [comment]
@jia200x please check the white space error

**Pull request merged**